commit b11c4d5bd9d3ccb07a55854ecf029b4d02418030
Author: Andrey Ulanov <andreyu@google.com>
Date:   Tue Mar 1 16:58:39 2016 -0800

    A couple of fixes in cloud-init systemd configs:
    
     - change all dependencies from network.target to network-online.target.
       All cloud-configs require network to be up.
     - Remove sshd.service dependencies.
     - wait for 'user-data' to be available before running cloud-init.service

diff --git a/systemd/cloud-config.service b/systemd/cloud-config.service
index 41a8614..d110b11 100644
--- a/systemd/cloud-config.service
+++ b/systemd/cloud-config.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Apply the settings specified in cloud-config
-After=network.target syslog.target cloud-config.target
+After=network-online.target syslog.target cloud-config.target
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/systemd/cloud-final.service b/systemd/cloud-final.service
index ef0f52b..7613d64 100644
--- a/systemd/cloud-final.service
+++ b/systemd/cloud-final.service
@@ -1,8 +1,8 @@
 [Unit]
 Description=Execute cloud user/final scripts
-After=network.target syslog.target cloud-config.service rc-local.service
+After=network-online.target syslog.target cloud-config.service rc-local.service
 Requires=cloud-config.target
-Wants=network.target
+Wants=network-online.target
 
 [Service]
 Type=oneshot
diff --git a/systemd/cloud-init.service b/systemd/cloud-init.service
index 6b0c722..d93fe20 100644
--- a/systemd/cloud-init.service
+++ b/systemd/cloud-init.service
@@ -1,12 +1,12 @@
 [Unit]
 Description=Initial cloud-init job (metadata service crawler)
-After=local-fs.target network.target cloud-init-local.service
-Before=sshd.service sshd-keygen.service
-Requires=network.target
-Wants=local-fs.target cloud-init-local.service sshd.service sshd-keygen.service
+After=network-online.target cloud-init-local.service
+Requires=network-online.target
+Wants=cloud-init-local.service
 
 [Service]
 Type=oneshot
+ExecStartPre=/usr/share/cloud/wait_for_user_data.sh
 ExecStart=/usr/bin/cloud-init init
 RemainAfterExit=yes
 TimeoutSec=0
diff --git a/systemd/cloud-init-local.service b/systemd/cloud-init-local.service
index a31985c..05cb5b9 100644
--- a/systemd/cloud-init-local.service
+++ b/systemd/cloud-init-local.service
@@ -1,7 +1,7 @@
 [Unit]
 Description=Initial cloud-init job (pre-networking)
-Wants=local-fs.target
-After=local-fs.target
+Wants=var-lib-cloud.mount
+After=var-lib-cloud.mount
 
 [Service]
 Type=oneshot
